src/main/java/seedu/address/model/util/DateUtil.java

```diff
@@ -132,44 +145,77 @@ public static boolean compareDates(Date date1, Date date2) {
         }
     }
 
+    /**
+     * generate a date that is 1 month behind current date
+     */
+    public static Date generateOutdatedDebtDate(Date currentDate) {
+        Calendar cal = Calendar.getInstance();
+        cal.setTime(currentDate);
+        cal.add(Calendar.MONTH, -1);
+        return cal.getTime();
+    }
+
+    /**
+     * Get the difference in number of months between the 2 dates.
+     * @param date1 is assumed to be AFTER date2.
+     * @return return difference. If there is no difference, return 0.
+     */
+    public static int getNumberOfMonthBetweenDates(Date date1, Date date2) {
+        Calendar cal1 = Calendar.getInstance();
+        cal1.setTime(date1);
+        Calendar cal2 = Calendar.getInstance();
+        cal2.setTime(date2);
+        // if date1 is only 1 year ahead of date2
+        if (cal1.get(Calendar.YEAR) - cal2.get(Calendar.YEAR) == 1) {
+            return cal1.get(Calendar.MONTH) + 1
+                    + (11 - cal2.get(Calendar.MONTH));
+        } else if (cal1.get(Calendar.YEAR) - cal2.get(Calendar.YEAR) > 1) {
+            return cal1.get(Calendar.MONTH) + 1
+                    + (11 - cal2.get(Calendar.MONTH))
+                    + (12 * (cal1.get(Calendar.YEAR) - cal2.get(Calendar.YEAR) - 1));
+        } else {
```

Comment:
same for this `else`

Reply:
Ok I see what you mean. But the else statements are required. If I have 3 `if` statements like what you are proposing, there is no guarantee that this method returns anything at all.